util/chplenv/chpl_tasks.py

```diff
@@ -13,10 +13,14 @@ def get():
     if not tasks_val:
         platform_val = chpl_platform.get('target')
         compiler_val = chpl_compiler.get('target')
+        arch_val = chpl_arch.get('target')
 
-        if (platform_val.startswith('cygwin') or
-                platform_val.startswith('netbsd') or
-                platform_val.startswith('freebsd')):
+        cygwin = platform_val.startswith('cygwin')
+        bsd = (platform_val.startswith('netbsd') or
+               platform_val.startswith('freebsd'))
+        mac_arm = platform_val.startswith('darwin') and arch_val == 'arm64'
```

Comment:
Writing my latest comment and looking at the code again, I found myself wondering:  Is there a principled answer to why/when we use `.startswith()` vs. `==` here?  `startswith` for `cygwin` makes sense since we support both `cygwin32` and `cygwin64`, I believe.  Grepping, it looks like the same is true for `netbsd` (though the logic for deciding that looks questionable to me... if we're running on anything other than `amd64`, we assume it's 32-bit netbsd?  Is that the only 64-bit architecture netbsd runs on?).  `darwin` and `freebsd` don't obviously have multiple flavors that I can tell by looking at our scripts or `uname` output (so maybe they are just defensive programming or pattern matching?).

I realize that most of these cases are pre-existing code, but I probably would use `==` on the new `darwin` test assuming we're not aware of `uname` returning different strings on different Mac generations.  [This](https://developer.apple.com/forums/thread/668206) makes it look as though it's not different on M1 Macs.  Grepping the sources,  it looks like most of our scripts use `==` with darwin with `chpl_unwind` being the only current exception.

Reply:
I had this question too, and I found that we're pretty inconsistent across our chplenv scripts about when we use `==` vs `startswith`. To me it seems like we're consistent within a file where if one thing uses startswith, all of them will so I just stuck with that here. (e.g. like you saw chpl_unwind uses `startsWith("darwin")`, but chpl_make uses `== "darwin"`).

I'm inclined to just leave things alone mostly so I don't have to retest, but if you feel strongly I will change it.

Reply:
I wouldn't say "strongly", but it also seems like a simple enough change that re-testing doesn't seem all that necessary or onerous to me.  That is, if it were my PR I'd do it, but I'm not going to be upset if you proceed as-is.